pkg/devicetemplateconverter/vendor_profile_fetcher.go

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package devicetemplateconverter
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
+	"go.thethings.network/lorawan-stack/v3/pkg/rpcmetadata"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+// VendorIDProfileFetcher handles the validation and fetching of end device profile in the Device Repository.
+type VendorIDProfileFetcher struct {
+	Component        Component
+	DeviceRepository ttnpb.DeviceRepositoryClient
+}
+
+// ShouldFetchProfile dictactes if the end device has the necessary fields to fetch its profile.
+func (*VendorIDProfileFetcher) ShouldFetchProfile(device *ttnpb.EndDevice) bool {
+	return device.GetVersionIds().GetVendorId() != 0
```

Comment:
`VendorProfileId` zero is not valid, so that should be checked as well, right ?

Reply:
Went based on this: https://github.com/TheThingsNetwork/lorawan-stack/pull/5610#discussion_r936595166

Will check when updating the other comments if its valid or not.

Reply:
It's not a valid value indeed. Only non-zero should be fetched.